partner_communication/models/communication_job.py

```diff
@@ -378,6 +370,25 @@ def write(self, vals):
         if vals.get("body_html") or vals.get("send_mode") == "physical":
             self.count_pdf_page()
 
+        # if the call must be done after the sending, we unlink all activities
+        # associated with the job (as for the moment, there is only one activity type)
+        if self.need_call == "after_sending" and self.state == "pending":
+            for activity in self.activity_ids:
+                activity.unlink()
```

Comment:
There is no need to loop through the records, you can directly call the method on the recordset.
```suggestion
            self.activity_ids.unlink()
```